### PR TITLE
Firebase CLI and local development

### PR DESCRIPTION
**Pull request by thompchr**
I reworked a couple of things in this PR. I removed all of the individual `package.json` files from the functions directories and created a top-level (within `functions`) `package.json` so the functions are all one module. Then each of the functions is exported as it's name. This _seems_ to a) work b) be what Google/Firebase recommends. It also allows us to use the Firebase CLI and emulators for local development. I also added a check within the `Firebase.tsx` file that configures Firebase to use the local emulators when running from `localhost`.

Getting the emulator configuration working:

1. `npm install -g firebase-tools` (or `npx firebase-tools` but Google recommends the `-g` route, not sure why)
1. Get an import from me on the data side of things to allow you to use your user. It still uses the same auth mechanism so your username/password are the same, but your "profile" is stored in firestore, so you need a data import. Place it in `./data`
1. Start the emulator with `npm run firebase`
1. Start the webpack dev server the normal way

Once it's running, you can then access the dev firebase console via `http://localhost:4000`

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+exports.funcTeacherSeqSummary = require('./funcTeacherSeqSummary').funcTeacherSeqSummary;
+exports.funcTeacherSeqTrend = require('./funcTeacherSeqTrend').funcTeacherSeqTrend;
+exports.funcTransitionLog = require('./funcTransitionLog').funcBehaviourTrend;
+exports.funcTransitionLogNew = require('./funcTransitionLogNew').funcTransitionLogNew;
+exports.funcTransitionSessionDates = require('./funcTransitionSessionDates').funcTransitionSessionDates;
+exports.funcTransitionSessionSummary = require('./funcTransitionSessionSummary').funcTransitionSessionSummary;
+exports.funcTransitionTrend = require('./funcTransitionTrend').funcBehaviourTrend;
+exports.funcTransitionTrend2 = require('./funcTransitionTrend2').funcBehaviourTrend;
+exports.funcTransitionTrendNew = require('./funcTransitionTrendNew').funcTransitionTrendNew;
+exports.funcTransitionTypeSummary = require('./funcTransitionTypeSummary').funcTransitionTypeSummary;
```
Comment (clarah2) [inline functions/index.js:55]: missing a few!

**Review by clarah2 — CHANGES_REQUESTED** [review]

**Review by clarah2 — DISMISSED** [review]
funcTeacherMathSummary has not been updated in GCP...the last deployed version was from April. Everything else looks good.

**Review by clarah2 — DISMISSED** [review]
awesome!

**Comment by thompchr** [comment]
Ok, rebased off of the latest changes. @clarah2 if you're good with this, you can merge once you submit a review!

**Review by clarah2 — APPROVED** [review]

**Pull request merged**